api/python/provisioner/commands/__init__.py

```diff
@@ -351,6 +351,139 @@ def run(self, *args, targets=ALL_MINIONS, **kwargs):
         return res
 
 
+@attr.s(auto_attribs=True)
+class SetNTP(CommandParserFillerMixin):
+    # TODO at least either pre or post should be defined
+    input_type: Type[
+        Union[inputs.ParamGroupInputBase, inputs.ParamDictItemInputBase]
+    ]
+    pre_states: List[State] = attr.Factory(list)
+    post_states: List[State] = attr.Factory(list)
+
+    _run_args_type = RunArgsUpdate
+
+    # TODO input class type
+    @classmethod
+    def from_spec(cls, input_type: str, states: Dict):
+        return cls(
+            input_type=getattr(inputs, input_type),
+            pre_states=[State(state) for state in states.get('pre', [])],
+            post_states=[State(state) for state in states.get('post', [])]
+        )
+
+    def _apply(self, params, targets):
+        pillar_updater = PillarUpdater(targets)
+
+        pillar_updater.update(params)
+        try:
+            logger.debug('Applying pre states')
```

Comment:
logger.info('Applying pre states for controller NTP')

Reply:
the states are for server node not for controller.